cli_tools/diagnostics/main_windows.go

```diff
@@ -34,6 +34,11 @@ const (
 	crashDump = `C:\Windows\MEMORY.dmp`
 )
 
+type winEvt struct {
```

Comment:
This could use a comment describing what it represents and what the "fromProvider" arg means.

Alternatively, instead of tracking a fromProvider arg, you could write the code below so that it tries the command with `-LogName` first, and if it fails then it tries `-ProviderName` instead. That would eliminate the need for the winEvt struct.

Reply:
Prefer adding comments here. The alternative seems a little complicate, as it might fail for other reasons, then we need to stop retry for those cases. And found that when the powershell cmd fails, we only got general `Exit status 1` error without any details, which seems interesting, need to figure out later if needed.